File: Python3/Medium/Accepted/validate-ip-address_3.py
class Solution:
    def validIPAddress(self, queryIP: str) -> str:
        
        if queryIP.count('.') == 3:
            nums = queryIP.split('.')
            for x in nums:
                if not x or len(x) > 3 or (x[0] == '0' and len(x) != 1) or not x.isdigit() or int(x) > 255:
                    return "Neither"
            return "IPv4"
        elif queryIP.count(':') == 7:
            nums = queryIP.split(':')
            hexdigits = '0123456789abcdefABCDEF'
            for x in nums:
                if not x or len(x) > 4 or not all(c in hexdigits for c in x):
                    return "Neither"
            return "IPv6"
        return "Neither"

File: C++/Medium/CompileError/validate-ip-address_3.cpp
class Solution {
public:
    string validIPAddress(string queryIP) {
        
        if (queryIP.find(".") != string::npos) {
            return isValidIPv4(queryIP) ? "IPv4" : "Neither";
        } else if (queryIP.find(":") != string::npos) {
            return isValidIPv6(queryIP) ? "IPv6" : "Neither";
        } else {
            return "Neither";
        }
    }
};